From: =?utf-8?q?Fabian_Gr=C3=BCnbichler?= <f.gruenbichler@proxmox.com>
Date: Mon, 24 Jun 2024 08:23:42 +0200
Subject: Revert "loongarch: add frecipe and relax target feature"

This reverts commit e81df3f3225ce53921a87b9e490959c41b783b02.

Debian: this commit breaks a lot of tests on loong64 since our rustc build is
still based on LLVM-17. This revert should be dropped once we switch over to
LLVM 18.

Bug: https://github.com/rust-lang/rust/issues/114661
---
 compiler/rustc_target/src/target_features.rs | 2 --
 tests/ui/abi/compatibility.rs                | 2 +-
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/compiler/rustc_target/src/target_features.rs b/compiler/rustc_target/src/target_features.rs
index 1b507bb..800907b 100644
--- a/compiler/rustc_target/src/target_features.rs
+++ b/compiler/rustc_target/src/target_features.rs
@@ -378,12 +378,10 @@ const LOONGARCH_ALLOWED_FEATURES: &[(&str, Stability)] = &[
     // tidy-alphabetical-start
     ("d", Unstable(sym::loongarch_target_feature)),
     ("f", Unstable(sym::loongarch_target_feature)),
-    ("frecipe", Unstable(sym::loongarch_target_feature)),
     ("lasx", Unstable(sym::loongarch_target_feature)),
     ("lbt", Unstable(sym::loongarch_target_feature)),
     ("lsx", Unstable(sym::loongarch_target_feature)),
     ("lvz", Unstable(sym::loongarch_target_feature)),
-    ("relax", Unstable(sym::loongarch_target_feature)),
     ("ual", Unstable(sym::loongarch_target_feature)),
     // tidy-alphabetical-end
 ];
diff --git a/tests/ui/abi/compatibility.rs b/tests/ui/abi/compatibility.rs
index 373d1cc..f77766b 100644
--- a/tests/ui/abi/compatibility.rs
+++ b/tests/ui/abi/compatibility.rs
@@ -39,7 +39,7 @@
 //@ revisions: loongarch64
 //@[loongarch64] compile-flags: --target loongarch64-unknown-linux-gnu
 //@[loongarch64] needs-llvm-components: loongarch
-//@[loongarch64] min-llvm-version: 18
+//@[loongarch64] min-llvm-version: 17
 //@ revisions: wasm
 //@[wasm] compile-flags: --target wasm32-unknown-unknown
 //@[wasm] needs-llvm-components: webassembly
